pkgs/development/compilers/swift/sourcekit-lsp/default.nix

```diff
@@ -0,0 +1,74 @@
+{ lib
+, stdenv
+, callPackage
+, fetchFromGitHub
+, swift
+, swiftpm
+, Foundation
+, XCTest
+, sqlite
+, ncurses
+, CryptoKit
+, LocalAuthentication
+}:
+let
+  # Generated by swiftpm2nix.
+  generated = callPackage ./generated { };
+
+  # On Darwin, we only want ncurses in the linker search path, because headers
+  # are part of libsystem. Adding its headers to the search path causes strange
+  # mixing and errors.
+  # TODO: Find a better way to prevent this conflict.
+  ncursesInput = if stdenv.isDarwin then ncurses.out else ncurses;
```

Comment:
Is this a general problem on Darwin or only for swift packages?

Reply:
I think this may be specific to the LLVM modules logic, because if it was broader, I feel like we would've encountered it elsewhere. If so, it's possible ObjC can also cause this (maybe even C? I believe the flag is `-fmodules`.)

Maybe we should eventually get rid of some of these headers in libsystem. It currently ships headers for ncurses, sqlite, krb5, maybe some others, but we package those separately (and libsystem doesn't even provide implementations of those). A similar discussion came up in: https://github.com/NixOS/nixpkgs/pull/186251#issuecomment-1213605428